### DISCUSSION TITLE
Unable to install compatible starlette version via Poetry

### DISCUSSION THREAD

**Discussion by jack-t-lange** [fastapi/fastapi#10248] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
dependencies = [
    "starlette>=0.27.0,<0.28.0",
    "pydantic>=1.7.4,!=1.8,!=1.8.1,!=2.0.0,!=2.0.1,!=2.1.0,<3.0.0",
    "typing-extensions>=4.5.0",
    # TODO: remove this pin after upgrading Starlette 0.31.1
    "anyio>=3.7.1,<4.0.0",
]
```

### Description

Per the code here:
https://github.com/tiangolo/fastapi/blob/bfde8f3ef20dc338bbce8a0ff0f4b441c54d5b64/pyproject.toml#L44

It appears that the most recent version of `FastAPI` is dependent on an older version of `starlette`. However, the notes for the most recent  `FastAPI` release `0.103.1` suggest that `starlette 0.31.1.` should be compatible.

Because Poetry tries to solve for version dependencies automatically, my FastAPI project is stuck using an older version of `starlette`. The older version of `starlette` `0.27.0` does not depend on the package `httpx`, so `httpx is never installed. However, the most recent version of FastAPI requires `httpx`

I was able to get my project running by manually installing the `httpx` package, but I am wondering if the version dependencies for FastAPI could be updated? I would be happy to open a PR if that is the appropriate next step.

### Operating System

Linux

### FastAPI Version

0.103.1

### Pydantic Version

2.3.0

### Python Version

3.10.12

**Comment by n8sty** [comment, marked as answer]
This note "Pin AnyIO to < 4.0.0 to handle an incompatibility while upgrading to Starlette 0.31.1" is communicating that _until_ FastAPI upgrades to Starlette 0.31.1 or greater the pint to AnyIO is needed, not that 103.1 of FastAPI is compatible with Starlette 0.31.1. There are some existing PRs already that attempt to upgrade Starlette, for eg: https://github.com/tiangolo/fastapi/pull/9636. Not saying you shouldn't open one, but I don't think the lack of a PR is necessarily the blocker here. Starlette also doesn't require `httpx`, it's installed as an extra dependency.

Assuming you need `httpx` for testing or anything else you can do what I do with my FastAPI projects that use Poetry for dependency management, just `poetry add httpx`. I believe this is what you already did, so just reiterating that there isn't anything else to do.
